pandas/core/indexes/base.py

```diff
@@ -506,6 +506,9 @@ def _shallow_copy(self, values=None, **kwargs):
         attributes.update(kwargs)
         if not len(values) and 'dtype' not in kwargs:
             attributes['dtype'] = self.dtype
+        from pandas import DatetimeIndex
```

Comment:
As discussed below, this code converts a DTI to ndarray before calling ``_simple_new``. All the other cases either seem to be an ndarray already or are converted to ndarray in the ``_simple_new`` function. I expect that there is probably a better way of handling this.

Reply:
you can just do

with a comment
``values = getattr(values, 'values', values)``